src/util/ThreadedMailbox.cc

```diff
@@ -225,19 +241,14 @@ namespace litecore { namespace actor {
 
 
     void ThreadedMailbox::runAsyncTask(void (*task)(void*), void *context) {
-        // std::future destructor blocks here, so can cause deadlock.  Need
-        // to keep it around
-        static uniform_int_distribution dis(1, INT32_MAX);
-        static map<int, future<void>> runningMap;
-        auto nextId = 0;
-        do {
-            nextId = dis(sRandGen);
-        } while(runningMap.find(nextId) != runningMap.end());
-
-        runningMap[nextId] = async(launch::async, [nextId, task](void* c) {
-            task(c);
-            runningMap.erase(nextId);
-        }, context);
+        static once_flag once;
```

Comment:
It's easier if you put the static var _inside_ the method; then it gets constructed the first time the method is called.
```
{
    static RunAsyncActor *sRunAsyncActor = retain(new RunAsyncActor());
    sRunAsyncActor->runAsync(task, context);
}
```